lib/src/metta/mod.rs

```diff
@@ -1,9 +1,12 @@
 //! Contains MeTTa specific types, constants and functions.
 
 pub mod text;
+#[cfg(feature = "old_interpreter")]
 pub mod interpreter;
```

Comment:
Thanks Luke, it is a nice point. I renamed `interpreter.rs` to `old_interpreter.rs` (see https://github.com/trueagi-io/hyperon-experimental/pull/728/commits/2bd75027e925c39e784b906398a5e479c42acb1d) but there is also part of `stdlib.rs` related to the old interpreter. I think it will be simpler to refactor this in a one step when old interpreter is being removed. Anyway `old_interpreter.rs` sounds more clear and aligned with `old_interpreter` feature.

I am still not sure if we need to rename `interpreter_minimal.rs` to the `interpreter.rs`. Minimal MeTTa is has evaluation order which is different from MeTTa evaluation order. I think in the future we could allow more flexible evaluation control in MeTTa and then minimal MeTTa is not needed. On the other hand main structure of the interpreter (except the set of instructions) probably will be the same. By this reason I would prefer keep it as `interpreter_minimal.rs` for a while.